File: decompiled/sources/com/hjq/base/RecyclerPagerAdapter.java
package com.hjq.base;

import android.view.View;
import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.PagerAdapter;

public final class RecyclerPagerAdapter extends PagerAdapter {
    private final RecyclerView.Adapter mAdapter;

    public boolean isViewFromObject(View view, Object obj) {
        return view == obj;
    }

    public RecyclerPagerAdapter(RecyclerView.Adapter adapter) {
        if (adapter != null) {
            this.mAdapter = adapter;
            adapter.registerAdapterDataObserver(new RecyclerView.AdapterDataObserver() {
                public void onChanged() {
                    RecyclerPagerAdapter.this.notifyDataSetChanged();
                }

                public void onItemRangeChanged(int i, int i2) {
                    RecyclerPagerAdapter.this.notifyDataSetChanged();
                }

                public void onItemRangeChanged(int i, int i2, Object obj) {
                    RecyclerPagerAdapter.this.notifyDataSetChanged();
                }

                public void onItemRangeInserted(int i, int i2) {
                    RecyclerPagerAdapter.this.notifyDataSetChanged();
                }

                public void onItemRangeRemoved(int i, int i2) {
                    RecyclerPagerAdapter.this.notifyDataSetChanged();
                }

                public void onItemRangeMoved(int i, int i2, int i3) {
                    RecyclerPagerAdapter.this.notifyDataSetChanged();
                }
            });
            return;
        }
        throw new IllegalArgumentException("are you ok?");
    }

    public int getCount() {
        return this.mAdapter.getItemCount();
    }

    public Object instantiateItem(ViewGroup viewGroup, int i) {
        RecyclerView.ViewHolder createViewHolder = this.mAdapter.createViewHolder(viewGroup, 0);
        viewGroup.addView(createViewHolder.itemView);
        this.mAdapter.onBindViewHolder(createViewHolder, i);
        return createViewHolder.itemView;
    }

    public void destroyItem(ViewGroup viewGroup, int i, Object obj) {
        viewGroup.removeView((View) obj);
    }
}

File: decompiled/sources/com/hjq/base/RecyclerPagerAdapter.kt
package com.hjq.base

import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.viewpager.widget.PagerAdapter

class RecyclerPagerAdapter(adapter: RecyclerView.Adapter<*>?) : PagerAdapter() {
    private val mAdapter: RecyclerView.Adapter<*>

    init {
        if (adapter != null) {
            mAdapter = adapter
            adapter.registerAdapterDataObserver(
                object : RecyclerView.AdapterDataObserver() {
                    override fun onChanged() {
                        notifyDataSetChanged()
                    }

                    override fun onItemRangeChanged(positionStart: Int, itemCount: Int) {
                        notifyDataSetChanged()
                    }

                    override fun onItemRangeChanged(
                        positionStart: Int,
                        itemCount: Int,
                        payload: Any?
                    ) {
                        notifyDataSetChanged()
                    }

                    override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
                        notifyDataSetChanged()
                    }

                    override fun onItemRangeRemoved(positionStart: Int, itemCount: Int) {
                        notifyDataSetChanged()
                    }

                    override fun onItemRangeMoved(fromPosition: Int, toPosition: Int, itemCount: Int) {
                        notifyDataSetChanged()
                    }
                }
            )
        } else {
            throw IllegalArgumentException("are you ok?")
        }
    }

    override fun isViewFromObject(view: View, obj: Any): Boolean {
        return view === obj
    }

    override fun getCount(): Int {
        return mAdapter.itemCount
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val adapter = mAdapter as RecyclerView.Adapter<RecyclerView.ViewHolder>
        val viewHolder = adapter.createViewHolder(container, 0)
        container.addView(viewHolder.itemView)
        adapter.onBindViewHolder(viewHolder, position)
        return viewHolder.itemView
    }

    override fun destroyItem(container: ViewGroup, position: Int, obj: Any) {
        container.removeView(obj as View)
    }
}
